temperature_regulator/server_dispatcher.py

```diff
@@ -0,0 +1,33 @@
+import socket
+from server_thread import ServerThread
+
+class ServerDispatcher:
+    LISTENER_SOCKET_ADDRESS = '127.0.0.1'
+    LISTENER_SOCKET_TIMEOUT = 1
+    MAX_CONNECTIONS = 10
+
+    def __init__(self, port: int):
+        self._listener_port = port
+        self._listener_socket = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM)
+        self._listener_socket.bind((ServerDispatcher.LISTENER_SOCKET_ADDRESS, self._listener_port))
+        self._listener_socket.listen(ServerDispatcher.MAX_CONNECTIONS)
+        self._listener_socket.settimeout(ServerDispatcher.LISTENER_SOCKET_TIMEOUT)
```

Comment:
Nie jestem pewien czy ten timeout jest tu zasadny bo serwer powinien zawsze czekać na nowe wiadomości, a w tym wypadku będzie co sekundę rzucał wyjatek i wracał do słuchania. Chyba, że stoi jeszcze inna logika za tym więc chętnie posłucham

Reply:
Timeout jest konieczny, aby serwer reagował na sygnały z zewnątrz i żeby można było zamknąć socket nasłuchujący z zewnątrz. Co jakiś czas serwer sprawdza czy jest sygnał. Jeśli nie, to wraca do blokady.

Reply:
Jeśli będzie to przydate to zrefaktoryzuję ten kod, aby jednocześnie serwer nasłuchiwał na socket jak i czekał na komendy, a gdy dostanie jedno z nich to wywoła odpowiedni callback. Zobaczę.